Source Generator Unique Hint Name Error

When using  a mixin on a nested class with the same name inside different parent classes.  I think Moxy generates the same "hint name"  and thus get this error.

```
CSC : error CS8785: Generator 'RoslynIncrementalGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'ArgumentExceptio
n' with message 'The hintName 'TimeWarp.Architecture.Features.TodoItems.Command.RouteMixin.Instance1.MixinCode.Moxy.g.cs' of the added source file must be unique within a generator. (Parameter 'hintName')'. [
D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Web.Contracts.csproj]
D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Features\Analytics\TrackEvent.cs(3,2): erro
r CS0246: The type or namespace name 'TrackEventValidiationAttribute' could not be found (are you missing a using directive or an assembly reference?) [D:\git\github\TimeWarpEngineering\timewarp-architecture\
Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Web.Contracts.csproj]
D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Features\Analytics\TrackEvent.cs(3,2): erro
r CS0246: The type or namespace name 'TrackEventValidiation' could not be found (are you missing a using directive or an assembly reference?) [D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\Ti
meWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Web.Contracts.csproj]
D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Features\TodoItems\Commands\RouteMixinNeste
dTestClass.cs(5,4): error CS0246: The type or namespace name 'RouteMixinAttribute' could not be found (are you missing a using directive or an assembly reference?) [D:\git\github\TimeWarpEngineering\timewarp-
architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Web.Contracts.csproj]
D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Features\TodoItems\Commands\RouteMixinNeste
dTestClass.cs(5,4): error CS0246: The type or namespace name 'RouteMixin' could not be found (are you missing a using directive or an assembly reference?) [D:\git\github\TimeWarpEngineering\timewarp-architect
ure\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Web.Contracts.csproj]
D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Features\TodoItems\Commands\UpdateTodoItem.
cs(5,4): error CS0246: The type or namespace name 'RouteMixinAttribute' could not be found (are you missing a using directive or an assembly reference?) [D:\git\github\TimeWarpEngineering\timewarp-architectur
e\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Web.Contracts.csproj]
D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Features\TodoItems\Commands\UpdateTodoItem.
cs(5,4): error CS0246: The type or namespace name 'RouteMixin' could not be found (are you missing a using directive or an assembly reference?) [D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\
TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Web.Contracts.csproj]
D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Features\TodoItems\Commands\CreateTodoItem.
cs(5,4): error CS0246: The type or namespace name 'RouteMixinAttribute' could not be found (are you missing a using directive or an assembly reference?) [D:\git\github\TimeWarpEngineering\timewarp-architectur
e\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Web.Contracts.csproj]
D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Features\TodoItems\Commands\CreateTodoItem.
cs(5,4): error CS0246: The type or namespace name 'RouteMixin' could not be found (are you missing a using directive or an assembly reference?) [D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\
TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Web.Contracts.csproj]
D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Features\TodoItems\Commands\CreateTodoItem.
cs(6,90): error CS0535: 'CreateTodoItem.Command' does not implement interface member 'IApiRequest.GetRoute()' [D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\tem
plates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Web.Contracts.csproj]
D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Features\TodoItems\Commands\CreateTodoItem.
cs(6,90): error CS0535: 'CreateTodoItem.Command' does not implement interface member 'IApiRequest.GetHttpVerb()' [D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\
templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Web.Contracts.csproj]
D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Features\TodoItems\TodoItemDto.cs(3,2): err
or CS0246: The type or namespace name 'CreateCommandAttribute' could not be found (are you missing a using directive or an assembly reference?) [D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\
TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Web.Contracts.csproj]
D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Features\TodoItems\TodoItemDto.cs(3,2): err
or CS0246: The type or namespace name 'CreateCommand' could not be found (are you missing a using directive or an assembly reference?) [D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.
Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Web.Contracts.csproj]
D:\git\github\TimeWarpEngineering\timewarp-architecture\Source\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Features\TodoItems\TodoItemDto.cs(3,17): er
ror CS0246: The type or namespace name 'UpdateCommandAttribute' could not be found (are you missing a using directive or an assembly reference?) [D:\git\github\TimeWarpEngineering\timewarp-architecture\Source
\TimeWarp.Architecture.Template\templates\TimeWarp.Architecture\Source\ContainerApps\Web\Web.Contracts\Web.Contracts.csproj]
```

--- a/Source/Lib/Morris.Moxy/SourceGenerators/ClassSourceGenerator.cs
+++ b/Source/Lib/Morris.Moxy/SourceGenerators/ClassSourceGenerator.cs
@@ -71,10 +71,11 @@ internal static class ClassSourceGenerator
 			return;
 
 		string? generatedSourceCode = null;
-		string classFileName = $"{classMeta.FullName}.{compiledTemplate.Name}.Instance{index}.MixinCode.Moxy.g.cs"
+		string classFileName = $"{classMeta.DeclaringTypeName}.{classMeta.ClassName}.{compiledTemplate.Name}.{index}.Moxy.g.cs"
 			.Replace("<", "{")
-			.Replace(">", "}");
-
+			.Replace(">", "}")
+			.TrimStart('.');
+		
 		try
 		{
 			generatedSourceCode = GenerateSourceCodeForAttributeInstance(productionContext, compiledTemplate, classMeta, attributeInstance);

--- a/Source/Lib/Morris.Moxy/SourceGenerators/ClassSourceGenerator.cs
+++ b/Source/Lib/Morris.Moxy/SourceGenerators/ClassSourceGenerator.cs
@@ -71,9 +71,10 @@ internal static class ClassSourceGenerator
 			return;
 
 		string? generatedSourceCode = null;
-		string classFileName = $"{classMeta.DeclaringTypeName}.{classMeta.ClassName}.{compiledTemplate.Name}.{index}.Moxy.g.cs"
+		string classFileName = $"{classMeta.Namespace}.{classMeta.DeclaringTypeName}.{classMeta.ClassName}.{compiledTemplate.Name}.{index}.Moxy.g.cs"
 			.Replace("<", "{")
 			.Replace(">", "}")
+			.Replace("..", ".")
 			.TrimStart('.');
 		
 		try

